#!/usr/bin/env python
PACKAGE = "ORB_SLAM3"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("Localization_mode", bool_t,   0, "Toggle localization mode on/off",  False)
gen.add("Reset_map", bool_t,   0, "Clears Atlas and active map",  False)
gen.add("Min_Neighbors",int_t, 0, "Minimum number of neighbors", 1, 10, 1000)
gen.add("Search_Radius", double_t, 0 , "Radius from point within which to search for neighbors",0.25, 0.0, 10.0)
gen.add("max_z", double_t, 0 , "Radius from point within which to search for neighbors",2.0, -10.0, 10.0)
gen.add("min_z", double_t, 0 , "Radius from point within which to search for neighbors",0.0, -10.0, 10.0)
gen.add("dist_thresh", double_t, 0 , "Maximum distance of a point from its keyframe",16.0, 0.0, 1000.0)
gen.add("free_thresh", int_t, 0 , "Minimum number of times a point is marked as free to be considered as actually free",5, 0, 65535)
gen.add("gaussian_ksize", int_t, 0 , "Kernel size for gaussian filter used to smooth occupancy grid counters",3, 0, 100)
gen.add("free_value", int_t, 0 , "amount to increase a cell's free value during raytracing",1, 1, 100)
gen.add("occupied_value", int_t, 0 , "amount to increase a cell's occupied value during raytracing",1, 1, 100)





exit(gen.generate(PACKAGE, "ORB_SLAM3", "orb_slam3"))
